client/my-sites/site-settings/controller.js

```diff
@@ -54,11 +54,36 @@ function renderPage( context, component ) {
 	);
 }
 
-module.exports = {
+const controller = {
 	redirectToGeneral() {
 		page.redirect( '/settings/general' );
 	},
 
+	redirectIfCantDeleteSite( context ) {
+		const state = context.store.getState();
+		const dispatch = context.store.dispatch;
+		const siteId = getSelectedSiteId( state );
+		const siteSlug = getSelectedSiteSlug( state );
+
+		if ( siteId && ! canDeleteSite( state, siteId ) ) {
+			return page.redirect( '/settings/general/' + siteSlug );
+		}
+
+		if ( ! siteId ) {
+			dispatch( {
+				type: SITES_ONCE_CHANGED,
+				listener: () => {
+					const updatedState = context.store.getState();
+					const updatedSiteId = getSelectedSiteId( updatedState );
+					const updatedSiteSlug = getSelectedSiteSlug( updatedState );
+					if ( ! canDeleteSite( updatedState, updatedSiteId ) ) {
```

Comment:
yes, we can, although, the expression will still evaluate to false without errors if the siteId is not there yet. As it happens right now when the data is not loaded at all